magento-engcom/import-export-improvements
# magento-engcom/import-export-improvements#101 Remove sku autogeneration on new products, keep on duplication

### Description

This PR may replace #102.
In contrast to the original PR I restored the SKU autogeneration in case a product is duplicated. The product copier will copy the original SKU and save the duplicate in a loop until a unique url key is found - if no SKU is autogenerated and no valid SKU is supplied it simply won't save the product.

Since the copier sets "isDuplicate" = true I thought it would be good simple solution to check for that entry before autogenerating a SKU.
### Fixed Issues (if relevant)


1. magento-engcom/import-export-improvements#101

### Manual testing scenarios


- Create product with sku "TEST"
- Create another product with sku "TEST"
- You should receive an error saying that the sku already exists instead of getting your sku modified.
- "Save & Duplicate" the product with sku "TEST"
- The duplicate receives the SKU "TEST-1" and will already be saved.


### Contribution checklist
